enhancements/windows-containers/csi-proxy.md

```diff
@@ -0,0 +1,189 @@
+workflow for adding a service to the expected service list.
+
+The csi-proxy binary will built from source and baked into the WMCO image. WMCO will copy the binary to each node at
+configuration time. A new item will be added to the windows-services ConfigMap, instructing the
+windows-instance-config-daemon to ensure that csi-proxy is always running as a Windows service on each node.
+
+With the nodes in this state, users are free to follow upstream documentation to deploy the CSI node drivers
+of their choice on the Windows nodes in the cluster.
+
+No support from Red Hat will be given for issues occurring within the upstream drivers, as they are not being
+distributed as part of OpenShift.
+
+### Workflow Description
+
+1. The cluster administrator will identify what CSI storage drivers the cluster requires for its workloads.
+2. Based on the requirements of the workloads, the cluster admin can follow upstream documentation to deploy the CSI
+   node drivers on its Windows nodes.
+
+### API Extensions
+
+N/A
+
+### Risks and Mitigations
+
+* If there are any security issues present in the CSI driver images deployed by the user, they will not be fixed
+  by upgrading OCP. It is up to the user to stay informed and update their images as necessary.
+* A mismatch between the OpenShift Linux CSI drivers and the user deployed Windows CSI drivers could cause undefined
+  behavior. This is described in the [version skew](#version-skew-strategy) section.
+
+### Drawbacks
+
+The main drawback to this approach is it puts the onus on the cluster administrator. As they will be required to deploy
+and maintain the Windows CSI node driver DaemonSet themselves, they will only be given support for issues with either
+the CSI driver controller, or the csi-proxy Windows service. Support will not be given for issues originating within
+the upstream CSI node drivers.
+
+This may cause an increased load on both support and developers, as support issues coming in will need to be
+properly filtered out based on the root cause of the problem.
+
+## Design Details
+
+### Test Plan
+
+WMCO already has an extensive end to end test suite. To facilitate testing the additions described in this enhancement,
+the e2e tests will be augmented to deploy a CSI node driver DaemonSet across Windows nodes, just as a user would. The
+e2e tests would then validate that a Windows pod is able to mount a volume backed by the driver. Initially vSphere
+storage will be tested, and other platforms can be tested in the future as desired. The purpose of testing is
+to ensure CSI proxy is properly interacting with the Windows instance and Windows pods.
+
+### Graduation Criteria
+
+#### Dev Preview -> Tech Preview
+
+N/A
+
+#### Tech Preview -> GA
+
+This feature is targeted for WMCO 9.0.0 in OpenShift 4.14.
+
+#### Removing a deprecated feature
+
+No action required.
+
+### Upgrade / Downgrade Strategy
```

Comment:
There should be a note how the csi-proxy will be upgraded on OCP upgrade. I guess you already have a process / description for a generic host service upgrade, a link would be appropriate.

Reply:
Expectation is that a grpc connection between a CSI driver and the csi proxy can get interrupted during upgrade. We expect that either the CSI driver re-tries the call or passes the error back to Kubernetes and Kubernetes retries.